webapp/tests/karma/unit/services/target-generator.js

```diff
@@ -398,4 +415,55 @@ describe('TargetGenerator service', function() {
     });
   });
 
+  it('gets the relevancy interval depending on settings', done => {
+    const settings = {
+      tasks: {
+        targets: {
+          items: [{ id: 'report', type: 'count' }]
+        }
+      }
+    };
+    Settings.resolves(settings);
+    UHCSettings.getMonthStartDate.returns(3);
+    UserContact.resolves();
+    RulesEngine.listen.callsArgWith(2, null, []);
+    injector.get('TargetGenerator')(() => {
+      chai.expect(UHCSettings.getMonthStartDate.callCount).to.equal(1);
+      chai.expect(UHCSettings.getMonthStartDate.args[0]).to.deep.equal([settings]);
+      chai.expect(CalendarInterval.getCurrent.callCount).to.equal(1);
+      chai.expect(CalendarInterval.getCurrent.args[0]).to.deep.equal([3]);
+      done();
+    });
+  });
+
+  it('ignores target instances outside if the relevancy interval', (done) => {
+    const settings = {
+      tasks: {
+        targets: {
+          items: [{ id: 'report', type: 'count' }]
+        }
+      }
+    };
+    Settings.resolves(settings);
+    UserContact.resolves();
+    UHCSettings.getMonthStartDate.returns(3);
+    CalendarInterval.getCurrent.returns({
```

Comment:
Would this scenario-based tests be more useful written as an integration test?  Instead of mocking CalendarInterval, consider using the two classes together.

Reply:
Sure. I removed the mock and kept a spy.